subsys/debug/asan.c

```diff
@@ -16,3 +16,32 @@ const char *__asan_default_options(void)
 #endif
 		"";
 }
+
+const char *__lsan_default_suppressions(void)
+{
+	return
+		/* SDL display driver
+		 *
+		 * The SDL2 library does not clean-up all it resources on exit,
+		 * as such suppress all memory leaks coming from libSDL2 and the
+		 * underlying X11 library
+		 */
+		"leak:libX11\n"
```

Comment:
* Would it be better to provide a suppression file and instructions on how to use it, instead of embedding this in the code?
* If it were to be in the code, would it not be better to ifdef this whole thing with the sdl display kconfig option?

Reply:
> Would it be better to provide a suppression file and instructions on how to use it, instead of embedding this in the code?

I have been thinking about this but in the end I opted for this approach as the ASAN related stuff is in one place.  Another advantage is that the user can provided his own suppression file without the need to be aware about the zephyr specific suppression file

> .If it were to be in the code, would it not be better to ifdef this whole thing with the sdl display kconfig option?

Agree, if we opt for the code base version I will update the PR